util/src/test/java/io/grpc/util/AdvancedTlsX509TrustManagerTest.java

```diff
@@ -147,6 +160,46 @@ public void clientTrustedWithSocketTest() throws Exception {
     assertEquals("No handshake session", ce.getMessage());
   }
 
+  @Test
+  public void updateTrustCredentials_rotate() throws GeneralSecurityException, IOException {
+    AdvancedTlsX509TrustManager trustManager = AdvancedTlsX509TrustManager.newBuilder().build();
+    trustManager.updateTrustCredentials(serverCert0File);
+    assertArrayEquals(serverCert0, trustManager.getAcceptedIssuers());
+
+    trustManager.updateTrustCredentials(serverCert0File, 1, TimeUnit.MINUTES,
+        executor);
+    assertArrayEquals(serverCert0, trustManager.getAcceptedIssuers());
+
+    fakeClock.forwardTime(1, TimeUnit.MINUTES);
```

Comment:
This caught a (minor) bug! I noticed `if (newTime == oldTime)` wasn't fully covered, so I thought we were missing a "time passes without an update" test. But, nope, it's here.

Within `updateTrustCredentials()` the `updateTime` is being lost, as it isn't being copied to `LoadFilePathExecution`. Let's pass the last update time to `LoadFilePathExecution`'s constructor so it won't needlessly reload the cert the first time the timer fires.

Reply:
1. Added test with: file content changed, file modification time is not changed
2. LoadFilePathExecution's constructor changed